simple/tests/stats/stat_var_hierarchy_generator_test.py

```diff
@@ -12,20 +12,82 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import json
+import os
+import shutil
+import tempfile
 import unittest
 
+import pandas as pd
 from stats.data import Triple
 from stats.stat_var_hierarchy_generator import *
+from stats.stat_var_hierarchy_generator import _generate_internal
+from tests.stats.test_util import is_write_mode
+
+_TEST_DATA_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                              "test_data", "stat_var_hierarchy_generator")
+_INPUT_DIR = os.path.join(_TEST_DATA_DIR, "input")
+_EXPECTED_DIR = os.path.join(_TEST_DATA_DIR, "expected")
+
+
+def _compare_files(test: unittest.TestCase, output_path, expected_path):
+  with open(output_path) as gotf:
+    got = gotf.read()
+    with open(expected_path) as wantf:
+      want = wantf.read()
+      test.assertEqual(got, want)
+
+
+def _read_triples_csv(path: str) -> list[Triple]:
+  df = pd.read_csv(path)
+  return [Triple(**kwargs) for kwargs in df.to_dict(orient='records')]
+
+
+def _test_generate_internal(test: unittest.TestCase, test_name: str):
+  test.maxDiff = None
+
+  with tempfile.TemporaryDirectory() as temp_dir:
+    input_triples_path = os.path.join(_INPUT_DIR, f"{test_name}.csv")
+
```

Comment:
Should we go from SV MCF to SV hierarchy JSON?  Its easier to just copy/paste MCFs from tests, prod, etc.

Reply:
Yes, will do in follow up PR.